semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -2038,19 +2067,22 @@ and map_universally_allowed_pattern (env : env)
         | Some tok -> (* "case" *) token env tok
         | None -> todo env ()
       in
-      let v2 =
-        match v2 with
-        | Some x -> map_user_type env x
-        | None -> todo env ()
-      in
       let v3 = (* dot_custom *) token env v3 in
-      let v4 = map_bound_identifier env v4 in
-      let v5 =
-        match v5 with
-        | Some x -> map_tuple_pattern env x
-        | None -> todo env ()
+      let id = map_bound_identifier env v4 in
+      let id_info = G.empty_id_info () in
```

Comment:
There should be helper functions we can use to avoid having to construct this manually. I don't remember what they are but let me know if you have trouble finding them and I can track them down.

Reply:
It seems I can't bind both to a single identifier with `name_of_id` because OCaml does not actually treat constructor arguments as tuples without explicit parens. It's very strange.